jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/table/BaseTableAssert.java

```diff
@@ -23,117 +23,150 @@ protected T table() {
+    public A size(int size, String... messages) {
+        return size(Matchers.is(size), messages);
+    }
+    
+    /**
+     * Match passed value with table rows count
+     * @param condition to compare
+     */
+    @JDIAction("Assert that '{name}' count {0}")
+    public A count(Matcher<Integer> condition) {
+        jdiAssert(table().count(), condition);
+        return (A) this;
     }
 
+    /**
+     * Match passed value with table rows count
+     * @param count to compare
+     */
+    @JDIAction("Assert that '{name}' count {0}")
+    public A count(int count) {
+        return count(Matchers.is(count));
+    }
+
+
     /**
      * Check that the table has the specified column
+     *
      * @param column to compare
      */
     @JDIAction("Assert that '{name}' has column'{0}'")
-    public A column(String column) {
-        jdiAssert(table().header(), hasItem(column));
+    public A column(String column, String... messages) {
+        jdiAssert(table().header(), hasItem(column), messages);
         return (A) this;
     }
-
+    
     /**
      * Check that the table has the specified columns
+     *
      * @param columns to compare
      */
     @JDIAction("Assert that '{name}' has columns '{0}'")
-    public A columns(List<String> columns) {
-        for(String column : columns)
-            column(column);
+    public A columns(List<String> columns, String... messages) {
+        for (String column : columns)
+            column(column, messages);
         return (A) this;
     }
-
+    
     /**
      * Match passed value with table columns
+     *
      * @param condition to compare
      */
     @JDIAction("Assert that '{name}' columns {0}")
-    public A columns(Matcher<Collection<? extends String>> condition) {
-        jdiAssert(table().header(), condition);
+    public A columns(Matcher<Collection<? extends String>> condition, String ... messages) {
+        jdiAssert(table().header(), condition, messages);
         return (A) this;
     }
+    
     /**
      * Check that the table has at list one specified row
+     *
      * @param matchers to compare
      */
     @JDIAction("Assert that '{name}' has at least one row that {0}}")
     public A rowThat(TableMatcher... matchers) {
         jdiAssert(TABLE_MATCHER.execute(table(), matchers), Matchers.is(not(Matchers.empty())));
         return (A) this;
     }
+    
     /**
      * Check that the table has at list one specified row
+     *
      * @param matcher to compare
      */
     @JDIAction("Assert that '{name}' has at least one row that {0}}")
-    public A rowThat(Single matcher, Column column) {
-        jdiAssert(TABLE_MATCHER.execute(table(), new TableMatcher[] {matcher.toTableMatcher(column)}), Matchers.is(not(Matchers.empty())));
+    public A rowThat(Single matcher, Column column, String ... messages) {
+        jdiAssert(TABLE_MATCHER.execute(table(), new TableMatcher[]{matcher.toTableMatcher(column)}), Matchers.is(not(Matchers.empty())), messages);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5133577)